File: JavaClients/guts-client-core/src/main/java/be/pxl/guts/core/models/AssignmentTestRun.java
package be.pxl.guts.core.models;

import java.util.List;

public class AssignmentTestRun {

    private Assignment assignment;
    private List<TestResult> results;
    private List<SolutionFile> solutionFiles;
    private String testCodeHash;

    public AssignmentTestRun(Assignment assignment, List<TestResult> results, List<SolutionFile> solutionFiles, String testCodeHash) {
        this.assignment = assignment;
        this.results = results;
        this.solutionFiles = solutionFiles;
        this.testCodeHash = testCodeHash;
    }

    public Assignment getAssignment() {
        return assignment;
    }

    public List<TestResult> getResults() {
        return results;
    }

    public List<SolutionFile> getSolutionFiles() {
        return solutionFiles;
    }

    public void setSolutionFiles(List<SolutionFile> solutionFiles) {
        this.solutionFiles = solutionFiles;
    }

    public String getTestCodeHash() {
        return testCodeHash;
    }

    public void setTestCodeHash(String testCodeHash) {
        this.testCodeHash = testCodeHash;
    }

}

File: DotNetClient/Guts.Client.Core/Models/AssignmentTestRun.cs
namespace Guts.Client.Core.Models;

public class AssignmentTestRun(
    Assignment assignment,
    IEnumerable<TestResult> results,
    IEnumerable<SolutionFile> solutionFiles,
    string testCodeHash)
{
    public Assignment Assignment { get;} = assignment;

    public IEnumerable<TestResult> Results { get; } = results;

    public IEnumerable<SolutionFile> SolutionFiles { get; } = solutionFiles;

    public string TestCodeHash { get; } = testCodeHash;
}